llm/memory_test.go

```diff
@@ -0,0 +1,127 @@
+package llm
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/ollama/ollama/api"
+	"github.com/ollama/ollama/envconfig"
+	"github.com/ollama/ollama/gpu"
+	"github.com/stretchr/testify/assert"
```

Comment:
Not critical for this PR but if only simple checks, it would be awesome to use `t.Fatal` as the rest of the codebase sticks as close to stdlib as possible

Reply:
I'll look at this in a follow up